lib/govuk_publishing_components/presenters/checkboxes_helper.rb

```diff
@@ -35,14 +36,21 @@ def initialize(options)
 
         @id = options[:id] || "checkboxes-#{SecureRandom.hex(4)}"
         @heading = options[:heading] || nil
-        @heading_size = options[:heading_size]
-        @heading_size = "m" unless %w[s m l xl].include?(@heading_size)
         @heading_caption = options[:heading_caption] || nil
+        @heading_size = options[:heading_size] || nil
         @is_page_heading = options[:is_page_heading]
         @description = options[:description] || nil
         @no_hint_text = options[:no_hint_text]
         @hint_text = options[:hint_text] || "Select all that apply." unless @no_hint_text
         @visually_hide_heading = options[:visually_hide_heading]
+
+        @size = if %w[s m l xl].include?(@heading_size)
```

Comment:
This could be split out into a separate method.

Reply:
Split this up into a separate method